src/battle-dex.ts

```diff
@@ -852,6 +852,38 @@ const Tools = {
 		return type;
 	},
 
+	getAbilitiesFor: function (template: any, gen: number) {
+		template = this.getTemplate(template);
+		if ((!isNaN(gen) && gen < 3) || !template.abilities) return {};
```

Comment:
`!isNaN`?

I assume the `gen` parameter is intended to be optional? What's the intended behavior when `gen` isn't passed? You probably want to specify a default gen, like:

`getAbilitiesFor: function (template: any, gen = 7) {`

or

`getAbilitiesFor: function (template: any, gen = 0) {`